src/app/researcher/study/request/[memberSlug]/study-proposal.tsx

```diff
@@ -5,19 +5,130 @@ import { Button, Group, Stack, Text } from '@mantine/core'
+import { StudyDocumentType } from '@/lib/types'
+import { zodResolver } from 'mantine-form-zod-resolver'
+import { CodeReviewManifest } from '@/lib/code-manifest'
+import { PresignedPost } from '@aws-sdk/s3-presigned-post'
+import {
+    signedUrlForCodeUploadAction,
+    signedUrlForDeletingStudyFilesAction,
+    signedUrlForStudyFileUploadAction,
+} from '@/server/actions/s3.actions'
+import { omit } from 'remeda'
+import { pathForStudy, pathForStudyDocuments } from '@/lib/paths'
 
-export const StudyProposal: React.FC<{ memberId: string }> = ({ memberId }) => {
+async function uploadFile(file: File, upload: PresignedPost) {
+    const body = new FormData()
+    for (const [key, value] of Object.entries(upload.fields)) {
+        body.append(key, value)
+    }
+    body.append('file', file)
+
+    const response = await fetch(upload.url, {
+        method: 'POST',
+        body,
+    })
+
+    if (!response.ok) {
+        notifications.show({
+            color: 'red',
+            title: 'Failed to upload file',
+            message: await response.text(),
+        })
+    }
+
+    return response.ok
+}
+
+async function uploadCodeFiles(files: File[], upload: PresignedPost, studyJobId: string) {
+    const manifest = new CodeReviewManifest(studyJobId, 'r')
+    for (const codeFile of files) {
+        manifest.files.push(codeFile)
+        await uploadFile(codeFile, upload)
+    }
+
+    const manifestFile = new File([manifest.asJSON], 'manifest.json', { type: 'application/json' })
+    return await uploadFile(manifestFile, upload)
+}
+
+export const StudyProposal: React.FC<{ memberSlug: string }> = ({ memberSlug }) => {
     const router = useRouter()
 
-    const { mutate: createStudy } = useMutation({
+    const studyProposalForm = useForm<StudyProposalFormValues>({
+        mode: 'uncontrolled',
+        validate: zodResolver(studyProposalFormSchema),
+        initialValues: {
+            title: '',
+            piName: '',
+            irbDocument: null,
+            descriptionDocument: null,
+            agreementDocument: null,
+            codeFiles: [],
+        },
+    })
+
+    const { isPending, mutate: createStudy } = useMutation({
         mutationFn: async (formValues: StudyProposalFormValues) => {
-            return await onCreateStudyAction({ memberId, studyInfo: formValues })
+            // Don't send any actual files to the server action, because they can't handle the file sizes
+            const valuesWithoutFiles = omit(formValues, [
+                'agreementDocument',
+                'descriptionDocument',
+                'irbDocument',
+                'codeFiles',
+            ])
+
+            const valuesWithFilenames = {
+                ...valuesWithoutFiles,
+                descriptionDocPath: formValues.descriptionDocument?.name || '',
+                agreementDocPath: formValues.agreementDocument?.name || '',
+                irbDocPath: formValues.irbDocument?.name || '',
+            }
```

Comment:
since the files are mandatory in the zod scheme, why would we not have filenames?  I think we can remove all the. `?.name || ''` bits, right?

Reply:
Maybe there is some cleaner way to approach this with zod, but the reason for this is:
- The zod schema is of type `File | null` because we can't seed the initial form values with an empty file for example. omitting it is effectively saying undefined too, so the reason we need this `?.` is because of the typing. i dont think its a huge deal but maybe theres a clever way around it simply using the zod schema, i spent a little time trying to figure that out but didn't come up with a solution

Reply:
oh, I see, that makes sense.  I wonder if we can use an empty file to make the initial values happy, I might test that out.

Reply:
yea ill try that out now and see - it might work